crates/bevy_ecs/src/change_detection.rs

```diff
@@ -372,10 +399,74 @@ where
     }
 }
 
-change_detection_impl!(Mut<'a, T>, T,);
+impl<'a, T> ComponentMut<'a> for Mut<'a, T> {
+    const ENABLED: bool = true;
+    fn build(
+        value: &'a mut Self::Inner,
+        added: &'a mut Tick,
+        changed: &'a mut Tick,
+        last_change_tick: u32,
+        change_tick: u32,
+    ) -> Self {
+        Mut {
+            value,
+            ticks: Ticks {
+                added,
+                changed,
+                last_change_tick,
+                change_tick,
+            },
+        }
+    }
+}
+
+change_detection_impl!(Mut<'a, T>, T, true,);
 impl_methods!(Mut<'a, T>, T,);
 impl_debug!(Mut<'a, T>,);
 
+impl<'a, T: ?Sized> DetectChanges for &'a mut T {
```

Comment:
This impl seems like a bit of a footgun. If you have a variable of type `&mut Mut<T>`, then it will implicitly use this no-op impl any time you call `.is_changed()`.